src/typed-method-types/apps.ts

```diff
@@ -208,6 +218,45 @@ export type DatastoreQueryResponse<
     items: DatastoreItem<Schema>[];
   };
 
+export type DatastoreCountArgs<
+  Schema extends DatastoreSchema,
+> =
+  & BaseMethodArgs
+  & {
```

Comment:
I think the arguments to `datastore.query` and `datastore.count` share a lot of fields, yes? Like `datastore`, `expression`, attributes and values.

I recommend factoring out these shared fields into its own dedicated `type` (doesn't have to be `export`ed, can just exist in this file), maybe something like a `type DynamoQueryArgs`, and have each of `DatastoreQueryArgs` and `DatastoreCountArgs` leverage that shared type.

This way, you don't duplicate all the work, and especially with JSDocs now being a part, don't have to duplicate the docs either!

Reply:
For an example of how to do this, see the Node Slack SDK here: https://github.com/slackapi/node-slack-sdk/blob/main/packages/web-api/src/types/request/conversations.ts#L52-L53

There are a bunch of non-exported `interface`s in the above file, and they're mixed and combined together using `|` and `&` in typescript for different `*Arguments` types (like `ConversationsAcceptSharedInviteArguments` that I linked to). It is essentially the solution to the same problem we are solving in this PR.